doc/user/content/sql/select.md

```diff
@@ -56,37 +65,74 @@ _join&lowbar;expr_ | A join expression; for more details, see our [`JOIN` docume
+additional intermediate state. To learn more about the costs of materialization,
+check out our [architecture overview](../../overview/architecture).
 
 ### Reading from sources and views
 
-Performing a `SELECT * FROM` a materialized source or materialized view is Materialize's ideal operation. When Materialize receives such a `SELECT` query, it quickly returns the maintained results from memory, with no re-execution.
+Performing a `SELECT * FROM` a materialized source or materialized view is
+Materialize's ideal operation. When Materialize receives such a `SELECT` query,
+it quickly returns the maintained results from memory, with no re-execution.
 
-Materialize also quickly returns results for queries that only filter, project, and re-order results of materialized sources or materialized views.
+Materialize also quickly returns results for queries that only filter, project,
+and re-order results of materialized sources or materialized views.
 
 ### Querying sources and views
 
-Performing a `SELECT` query that does not directly read out of a materialization requires Materialize to evaluate your query. Materialize will construct a temporary dataflow to materialize your query, and remove the dataflow as soon as it returns the query results to you.
+Performing a `SELECT` query that does not directly read out of a materialization
+requires Materialize to evaluate your query. Materialize will construct a
+temporary dataflow to materialize your query, and remove the dataflow as soon as
+it returns the query results to you.
 
-When you perform a `SELECT` query, all of your inputs must be *queryable*. An input is queryable if it is a constant collection, materialized, or it depends only on queryable inputs itself. The main reason an input might not be queryable is it if relies on an unmaterialized source for its definition.
+When you perform a `SELECT` query, all of your inputs must be *queryable*. An
+input is queryable if it is a constant collection, materialized, or it depends
+only on queryable inputs itself. The main reason an input might not be queryable
+is it if relies on an unmaterialized source for its definition.
 
-You can create a materialized view out of any query using [`CREATE MATERIALIZED VIEW`](../create-materialized-view), and it will always then be queryable.
+You can create a materialized view out of any query using [`CREATE MATERIALIZED
+VIEW`](../create-materialized-view), and it will always then be queryable.
 
-If you supply an `AS OF <time>` argument to your `SELECT` query the queryable requirement is lifted.
+If you supply an `AS OF <time>` argument to your `SELECT` query the queryable
+requirement is lifted.
+
+### Common table expressions (CTEs)
+
+Common table expressions, also known as CTEs and `WITH` queries, create aliases
+for statements that subsequent expressions can refer to (including subsquent
+CTEs). This can enhance legibility of complex queries, but doesn't alter the
+query's semantics.
```

Comment:
nit: changing plurality of "queries". I think either "This can enhance legibility of a complex query, but does not alter the query's semantics" or "This can enhance legibility of complex queries, but doesn't alter the queries' semantics."?